vowpalwabbit/parse_regressor.cc

```diff
@@ -305,36 +291,27 @@ void save_load_header(
         if (all.model_file_ver >=
             VERSION_FILE_WITH_INTERACTIONS)  // && < VERSION_FILE_WITH_INTERACTIONS_IN_FO (previous if)
         {
+          if (!read)
+            THROW("cannot write legacy format");
+
           // the only version that saves interacions among pairs and triples
-          uint32_t len = (uint32_t)all.interactions.size();
+          uint32_t len = 0;
 
           msg << len << " interactions: ";
           bytes_read_write +=
               bin_text_read_write_fixed_validated(model_file, (char*)&len, sizeof(len), "", read, msg, text);
 
           for (size_t i = 0; i < len; i++)
           {
+            // Only the read path is implemented since this is for old version read support.
             uint32_t inter_len = 0;
-            if (!read)
-            {
-              inter_len = (uint32_t)all.interactions[i].size();
-              msg << "len: " << inter_len << " ";
-            }
             bytes_read_write += bin_text_read_write_fixed_validated(
                 model_file, (char*)&inter_len, sizeof(inter_len), "", read, msg, text);
-            if (!read)
-            {
-              memcpy(buff2, all.interactions[i].data(), inter_len);
-
-              msg << "interaction: ";
-            // Copies data to stringstream regardless of existence of null characters.
-            // This might result in unintuitive behavior i.e. copy data after nulls as well.
-              msg.write(reinterpret_cast<char*>(all.interactions[i].data()), inter_len);
-            }
 
             bytes_read_write += bin_text_read_write_fixed_validated(model_file, buff2, inter_len, "", read, msg, text);
 
-            if (read)
+            std::vector<namespace_index> temp(buff2, buff2 + inter_len);
```

Comment:
buf2 isn't guaranteed to be `inter_len` in in size is it? Fewer bytes could have been read. To be accurate here, it would have to be
```
auto size = bin_text_read_write_fixed_validated(model_file, buff2, inter_len, "", read, msg, text);
bytes_read_write += size;
std::vector<namespace_index> temp(buff2, buff2 + size);
```

Reply:
If bytes read is less than the expected size, then it really should be an error here